src/credentials/auth.ts

```diff
@@ -701,6 +709,16 @@ export class Auth implements AuthService, ConnectionManager {
             return
         }
 
+        // Clear anything stuck in an 'authenticating...' state
+        // This can rarely happen when closing VS Code during authentication
+        await Promise.all(
```

Comment:
[Promise.allSettled](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/allSettled) may be useful here if we want to handle errors.

Reply:
Yeah it'll throw when updating a profile that no longer exists. Which could only happen in some sort of race condition. At least here only `vscode.Memento.update` could throw as all `getProfileOrThrow` calls happen within the same event loop.

A safer and more robust solution might be to store all state in-mem and add an 'uninitialized' state so it can be lazily eval'd as-needed. I'll try making that change when we address "linked accounts"